apps/sms/js/compose.js

```diff
@@ -741,46 +741,64 @@ var Compose = (function() {
     },
 
     onAttachmentClick: function thui_onAttachmentClick(event) {
-      if (event.target.classList.contains(attachmentClass) && !state.resizing) {
-        this.currentAttachmentDOM = event.target;
-        this.currentAttachment = attachments.get(event.target);
-        AttachmentMenu.open(this.currentAttachment);
+      if (!event.target.classList.contains(attachmentClass) || state.resizing) {
+        return;
       }
-    },
 
-    onAttachmentMenuClick: function thui_onAttachmentMenuClick(event) {
-      event.preventDefault();
-      switch (event.target.id) {
-        case 'attachment-options-view':
-          this.currentAttachment.view();
-          break;
-        case 'attachment-options-remove':
-          dom.message.removeChild(this.currentAttachmentDOM);
-          disposeAttachmentNode(this.currentAttachmentDOM);
+      this.currentAttachmentDOM = event.target;
+      this.currentAttachment = attachments.get(event.target);
 
-          state.size = null;
-          onContentChanged();
-          AttachmentMenu.close();
-          break;
-        case 'attachment-options-replace':
-          var request = this.requestAttachment();
-          request.onsuccess = (function replaceAttachmentWith(newAttachment) {
-            var fragment = insert(newAttachment);
+      var name = this.currentAttachment.name;
+      var blob = this.currentAttachment.blob;
+      var options = {
+        header: name.substr(name.lastIndexOf('/') + 1)
+      };
 
-            dom.message.insertBefore(fragment, this.currentAttachmentDOM);
+      // Localize the name of the file type
+      var mimeFirstPart = blob.type.substr(0, blob.type.indexOf('/'));
+      var fileType = (TYPES.indexOf(mimeFirstPart) > -1) ?
+        mimeFirstPart: 'others';
```

Comment:
typo: ```other``` ---> we have l10n label for ```view-attachment-other```.

Not related to this patch, I'm wondering why we don't have separate label for contact and treat it as ```other```?